app/hooks/use-hide-amount.ts

```diff
@@ -0,0 +1,21 @@
+import { useHideBalanceQuery } from "@app/graphql/generated"
+import { makeVar, useReactiveVar } from "@apollo/client"
```

Comment:
I think this is the only place where we are using `makeVar` and `useReactiveVar` in the application. If we wanted to prevent adding a new patter, I think similar behavior could be accomplished with a context provider like `TemporaryHideAmountContext`.